src/graph/UndirectedGraph.java

```diff
@@ -30,12 +30,16 @@ public Edge removeEdge(Edge e) {
     return end.removeNeighbor(e.getBegin());
   }
 
-  public DirectedGraph toDirected() {
-    DirectedGraph g = new DirectedGraph();
+  @Override
+  public Object clone() {
```

Comment:
I do not get this. Is there a difference between cloning directed and undirected graph? I think they shall be the same. Just clone the vertices, right?

Reply:
The problem I am facing now is that two vertices share the  same Edge object in undirected graphs, therefore deep clone every vertices will break that property. How can I address this?

Reply:
O. I see. You can do it like this then.